app/src/main/java/io/github/iurimenin/upcomingmovies/presenter/ListMoviesTask.kt

```diff
@@ -16,24 +16,38 @@ import java.net.HttpURLConnection
 import java.net.URL
 import java.util.*
 import kotlin.collections.ArrayList
+import kotlin.collections.HashMap
 
 
 /**
  * Created by Iuri Menin on 26/08/17.
  */
-class ListMoviesTask(private val delegate: AsyncTaskDelegate?, private val context: Context) : AsyncTask<String, Void, ArrayList<MovieVO>>() {
+class ListMoviesTask(private val mCallback: AsyncTaskCallback,
```

Comment:
This class is doing too much: it is abstracting background processing; doing network calls; persisting data. Can you see how to improve it?

Reply:
Yes, it would be best to separate into other classes only responsible for each of the functions (network calls, persisting data, etc)